src/sonic-host-services/host_modules/showtech.py

```diff
@@ -0,0 +1,55 @@
+"""Show techsupport command handler"""
+
+import host_service
+import subprocess
+import re
+
+MOD_NAME = 'showtech'
+
+class Showtech(host_service.HostModule):
+    """DBus endpoint that executes the "show techsupport" command
+    """
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='s', out_signature='is')
+    def info(self, date):
+
+        ERROR_TAR_FAILED = 5
+        ERROR_PROCFS_SAVE_FAILED = 6
+        ERROR_INVALID_ARGUMENT = 10
+
+        err_dict = {ERROR_INVALID_ARGUMENT: 'Invalid input: Incorrect DateTime format',
+                    ERROR_TAR_FAILED: 'Failure saving information into compressed output file',
+                    ERROR_PROCFS_SAVE_FAILED: 'Saving of process information failed'}
+
+        cmd = ['/usr/local/bin/generate_dump']
+        if date:
+            cmd.append("-s")
+            cmd.append(date)
+
+        try:
+            rc = 0
+            output = subprocess.check_output(cmd)
+
+        except subprocess.CalledProcessError as err:
+            rc = err.returncode
+            errmsg = err_dict.get(rc)
+
+            if errmsg is None:
+                output = 'Error: Failure code {:-5}'.format(rc)
+            else:
+                output = errmsg
+
+            print("%Error: Host side: Failed: " + str(rc))
+            return rc, output
+
+        output_string = output.decode("utf-8")
+        output_file_match = re.search('\/var\/.*dump.*\.gz', output_string)
+        if output_file_match is not None:
```

Comment:
> is not None [](http://example.com/codeflow?start=28&length=12)

You can remove them.

Reply:
If the user specifies an invalid date/time, the output will provide the appropriate error message instead of a filename. In that case, the regex match will be None. Other errors could also result in failure to provide an output file and filename.

Reply:
I have deleted the pattern match checking for the "success" case.